File: 7.js
// find 10001 prime number
function isPrime(n){
  for(var i = 2; i <= Math.ceil(Math.sqrt(n)); i++){
    if(n % i === 0 && n != i){
      return false;
    }
  }
  return true;
}

function main(n){
  // find n'th prime number
  var count = 0;
  var curNum = 1;

  while(n != count){
    curNum++;
    if(isPrime(curNum)){
      count++;
    }
  }
  return curNum;
}
console.log(isPrime(2));
console.log(main(10001));

File: 7.cpp
#include <cmath>
#include <iostream>
using namespace std;

bool isPrime(long long n) {
  for(long long i = 2; i <= ceil(sqrt(n)); i++) {
    if(n % i == 0 && n != i) {
      return false;
    }
  }
  return true;
}

int main() {
//cout << boolalpha << isPrime(77) << isPrime(17) << endl;
  long long count = 0;
  long long curNum = 1;

  while(count != 10001) {
    curNum++;
    if(isPrime(curNum)) {
      count++;
    }
  }
  cout << curNum << endl;
}




